### ISSUE TITLE
ENH: feature request for initial hessian estimate in scipy bfgs routine

### ISSUE THREAD

**Issue by danleonte** [scipy/scipy#17864]
### Is your feature request related to a problem? Please describe.

At the moment `scipy.optimize.minimize(..., method = 'BFGS')` does not accept an initial estimate for the inverse of the Hessian. This would be helpful because

1) it would make the algorithm more efficient when a good estimate is available
2) it allows using information from a previous second-order optimization routine which failed to converge because of the max number of iterations being reached. 

### Describe the solution you'd like.

It is easy to change 

`    I = np.eye(N, dtype=int)
    Hk = I` 

from lines 1373 - 1384 of [bfgs_implementation](https://github.com/scipy/scipy/blob/v1.10.0/scipy/optimize/_optimize.py#L1189-L1315)

to 

`    Hk = Hk_initial_estimate`

where  `Hk_initial_estimate` is a parameter to be passed by the user, and which is by default set to `np.eye(N, dtype = int)`. It can also be easily checked if the initial estimate is positive definite and symmetric.

### Describe alternatives you've considered.

The jax.scipy and tfp bfgs routines hang, hence I have resorted to the scipy implementation and only estimate the function and the gradients with jax on a gpu.  This would also benefit users of jax and tfp.

**Comment by dschmitz89** [comment]
This sounds like a useful feature. Would you be interested in opening a PR?

**Comment by paulestano** [comment]
Is there anybody working on this? I have some code ready if that's still needed.

**Comment by dschmitz89** [comment]
There is no open PR, so you are very welcome to submit one 👍 .
